src/decode-ethernet.c

```diff
@@ -39,7 +39,7 @@
 #include "util-debug.h"
 
 int DecodeEthernet(ThreadVars *tv, DecodeThreadVars *dtv, Packet *p,
-                   uint8_t *pkt, uint16_t len, PacketQueue *pq)
+                   uint8_t *pkt, uint32_t len, PacketQueue *pq)
```

Comment:
since we're passing length as "len - ETHERNET_HEADER_LEN" to the DecodeIPV4 func, it seems we can now get into trouble by sending > USHRT_MAX. I think this will lead to a int overflow / wrap around. Think we need explicit checks to avoid this case?